src/main/kotlin/no/nav/syfo/aktivitetskrav/cronjob/AktivitetskravAutomatiskOppfyltCronjob.kt

```diff
@@ -0,0 +1,54 @@
+package no.nav.syfo.aktivitetskrav.cronjob
+
+import net.logstash.logback.argument.StructuredArguments
+import no.nav.syfo.aktivitetskrav.AktivitetskravService
+import no.nav.syfo.application.cronjob.Cronjob
+import no.nav.syfo.application.cronjob.CronjobResult
+import no.nav.syfo.application.database.DatabaseInterface
+import org.slf4j.LoggerFactory
+import java.util.UUID
+
+class AktivitetskravAutomatiskOppfyltCronjob(
+    private val database: DatabaseInterface,
+    private val aktivitetskravService: AktivitetskravService,
+) : Cronjob {
+    override val initialDelayMinutes: Long = 5
+    override val intervalDelayMinutes: Long = 240
+
+    override suspend fun run() {
+        val result = runJob(uuids.map { UUID.fromString(it) })
+        log.info(
+            "Completed aktivitetskrav automatisk oppfylt processing job with result: {}, {}",
+            StructuredArguments.keyValue("failed", result.failed),
+            StructuredArguments.keyValue("updated", result.updated),
+        )
+    }
+
+    fun runJob(aktivitetskravUuids: List<UUID>): CronjobResult {
+        val result = CronjobResult()
+
+        val aktivitetskravList = aktivitetskravUuids.mapNotNull { aktivitetskravService.getAktivitetskrav(it) }
+        try {
+            database.connection.use { connection ->
+                aktivitetskravList.forEach {
+                    aktivitetskravService.oppfyllAutomatisk(
```

Comment:
Kunne kanskje sjekke om status allerede er AUTOMATISK_OPPFYLT? 

Det kan jo være det skjer litt krøll som gjør at cronjob'en kjører mer enn en gang.

Reply:
Godt poeng. Tror ikke det gjør så mye om det oppdateres til AUTOMATISK_OPPFYLT flere ganger, men kan godt legge det til for sikkerhets skyld.